tests/test_searchspace.py

```diff
@@ -281,3 +281,22 @@ def test_cardinality_constraints_with_overlapping_parameters():
                 ),
             ),
         )
+
+
+def test_cardinality_constraint_with_invalid_parameter_bounds():
+    """Impose a cardinality constraint on a parameter whose valid area does not
+    include zero raises an error."""  # noqa
+    parameters = (
+        NumericalContinuousParameter("c1", (0, 1)),
+        NumericalContinuousParameter("c2", (1, 2)),
+    )
+    with pytest.raises(ValueError, match="must include zero"):
```

Comment:
See other comment: It is possible to fulfill the cardinality constraint in this example, since. `c1` can be set to 0, so I am not sure if this should be an invalid configuration.

Reply:
You are right that this is not a restriction mathematically. But I think it is a reasonable thing to request from the user, mostly for three reasons:
* If a parameter cannot take the value 0, it has no influence on the cardinality constraint at all, i.e. no matter how you set its value, it will always increase the cardinality of the set by 1 because its always active. Therefore, you can always just rewrite the constraint without the parameter and with cardinality limits reduced by one.
* If we do the above and always expect that we work with a "non-redundant" constraint representation, it makes our lives easier implementation-wise
* To avoid unintended user input: perhaps they simply didn't notice that some of their parameters are completely irrelevant for the cardinality constraint as described above. This way, we automatically inform the user about the "unnecessary" input, and perhaps they then realize that they haven't fully thought their setup through.

@Scienfitz: Opinions?

Reply:
Ok, if we make this clear for the user this is fine for me.